watcher/watcher.go

```diff
@@ -104,51 +104,54 @@ func (c *ControllerInfo) ContactLightfoot() {
 	}
 }
 
+func CheckPodLabels(labels map[string]string) bool {
+	for k, v := range labels {
+		if k == "lightfoot" && v == "enable" {
+			return true
+		}
+	}
+	return false
+}
+
+func AddPod(name string, p PodInfo) {
+	controllerInfo.container[name] = p
+	controllerInfo.mu.Lock()
+	controllerInfo.pending[name] = true
+	controllerInfo.mu.Unlock()
+	controllerInfo.wake <- struct{}{}
+}
+
 func handlePodEvent(e PodEvent, pod *v1.Pod) {
 	switch e {
 	case Add:
-		proceed := false
 		// Add the ip to the lightfoot container corresponding to nodename
 		name := pod.ObjectMeta.GetName()
 		log.Println("Adding ", name)
 		if strings.HasPrefix(name, "lightfoot-daemon") {
 			controllerInfo.nodes[pod.Spec.NodeName] = pod.Status.PodIP
 			break
 		}
-		for k, v := range pod.ObjectMeta.GetLabels() {
-			if k == "lightfoot" && v == "enable" {
-				proceed = true
-				break
-			}
+		if !CheckPodLabels(pod.ObjectMeta.GetLabels()) {
+			break
```

Comment:
could the label be deleted from a running pod? In this case we should stop debugging, right?

Reply:
That's right. I don't have code in c++ for that yet.